src/provider/mod.rs

```diff
@@ -901,14 +907,22 @@ async fn handle_stream(
         }
         None => {
             debug!("not found {}", hash);
-            notify_transfer_aborted(events, connection_id, request_id);
-            writer.finish().await?;
+            notify_transfer_aborted(&writer);
+            writer.inner.finish().await?;
         }
     };
 
     Ok(())
 }
 
+/// A helper struct that combines a quinn::SendStream with auxiliary information
+struct ResponseWriter {
+    inner: quinn::SendStream,
+    events: broadcast::Sender<Event>,
+    connection_id: u64,
```

Comment:
I'd prefer not to include this here since it's directly available in the `inner` already.

Reply:
I just copied this from how it was before, the rationale being that the connection_id might not always be just what it currently is.

But we could make connection_id a fn that grabs it from the SendStream for now?